test/automation/src/sql/connectionDialog.ts

```diff
@@ -31,7 +31,9 @@ export class ConnectionDialog extends Dialog {
 	private static readonly CONNECT_BUTTON_SELECTOR = '.modal .modal-footer a[aria-label="Connect"]:not(.disabled)';
 	async connect(): Promise<void> {
 		await this.code.waitAndClick(ConnectionDialog.CONNECT_BUTTON_SELECTOR);
-
-		return this.waitForDialogGone();
+		// Try pressing the connect button again if the dialog does not go away
+		await this.waitForDialogGone().catch(async _ =>
```

Comment:
With the additional logging I think we should remove this for the time being until we figure out what's actually going on.